packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -0,0 +1,1144 @@
+    Key? key,
+    this.animationDuration = const Duration(milliseconds: 500),
+    this.selectedIndex = 0,
+    required this.destinations,
+    this.onDestinationSelected,
+    this.elevation = 8,
+    this.backgroundColor,
+    this.height,
+    this.labelBehavior = NavigationBarDestinationLabelBehavior.alwaysShow,
+  }) : super(key: key);
+
+  /// Determines how long the animation is for each destination as it goes from
+  /// unselected to selected.
+  final Duration animationDuration;
+
+  /// Determines which destination from [destinations] is currently selected.
+  ///
+  /// When this is updated, the destination (from [destinations]) at
+  /// [selectedIndex] goes from unselected to selected. It will animate its
+  /// [NavigationBarDestinationInfo.selectedAnimation] from 0.0 to 1.0.
+  final int selectedIndex;
+
+  /// The list of destinations (usually [NavigationBarDestination]s) in this
+  /// [NavigationBar].
+  ///
+  /// When [selectedIndex] is updated, the destination from this list at
+  /// [selectedIndex] will animate from 0 (unselected) to 1.0 (selected). When
+  /// the animation is increasing or completed, the destination is considered
+  /// selected, when the animation is decreasing or dismissed, the destination
+  /// is considered unselected.
+  ///
+  /// Any widget in this list will have access to a
+  /// [NavigationBarDestinationInfo] inherited widget.
+  final List<Widget> destinations;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// This callback usually updates the int passed to [selectedIndex].
+  final ValueChanged<int>? onDestinationSelected;
+
+  /// The z-coordinate of this [NavigationBar].
+  final double elevation;
+
+  /// The color of the [NavigationBar] itself.
+  ///
+  /// If null, defaults to [ColorScheme.surface].
+  final Color? backgroundColor;
+
+  /// The height of the [NavigationBar]'s Material, from the bottom of the
+  /// screen to the top border of the widget, not including safe area padding
+  /// for things like system navigation.
+  ///
+  /// Defaults to 74 when labels are used (when [labelBehavior] is
+  /// [NavigationBarDestinationLabelBehavior.alwaysShow] or
+  /// [NavigationBarDestinationLabelBehavior.onlyShowSelected]). Defaults to 56
+  /// when no labels are used (when [labelBehavior] is
+  /// [NavigationBarDestinationLabelBehavior.alwaysHide]).
+  final double? height;
+
+  /// Determines the behavior for how the labels will layout.
+  ///
+  /// Can be used to show all labels (the default), show only the selected
+  /// label, or hide all labels.
+  final NavigationBarDestinationLabelBehavior labelBehavior;
+
+  @override
+  Widget build(BuildContext context) {
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final double effectiveHeight = height ?? 80;
+    return Material(
+      elevation: elevation,
+      color: backgroundColor ?? ElevationOverlay.colorWithOverlay(colorScheme.surface, colorScheme.onSurface, 3.0),
```

Comment:
There should be a comment sort of explaining why the default background color using an overlay, something about Material You